lib/log_displayer.js

```diff
@@ -41,13 +42,27 @@ function colorize (line) {
 
 function readLogs (logplexURL) {
   return new Promise(function (resolve, reject) {
-    let res = cli.got.stream(logplexURL)
-    res.setEncoding('utf8')
-    liner.setEncoding('utf8')
-    res.pipe(liner)
-    liner.on('data', line => cli.log(colorize(line)))
-    res.on('end', resolve)
-    res.on('error', reject)
+    let u = url.parse(logplexURL, true)
+    let isTail = u.query.tail && u.query.tail === 'true'
+    let es = new EventSource(logplexURL)
+
+    es.onerror = function (err) {
+      if (!isTail) {
+        resolve()
+        es.close()
+      }
+
+      if (err && (err.status === 404 || err.status === 403)) {
+        reject(err)
+        es.close()
+      }
+    }
+
+    es.onmessage = function (e) {
+      e.data.split(/\n+/).forEach((line) => {
```

Comment:
It looks like the lines come in big chunks rather than one-message-per-log-line--I don't know if this is the right way to split them. I tried splitting on `\n` initially, but it looks like there are two newlines between each log line--I don't really understand why but this works around that.

Reply:
I didn't know you could pass a regex into split

Reply:
I don't know if this is helpful, sounds like maybe the opposite of what you want, but we have this in the project that splits a stream into lines: https://github.com/heroku/heroku-run/blob/master/lib/line_transform.js

Reply:
As far as I can tell, the `.data` property [of the message](https://developer.mozilla.org/en-US/docs/Web/Events/message) is a String--I don't think there's much point turning it back into a stream to process it like that, right? I haven't used SSE much, and never in node.

Reply:
And fwiw, I actually drop this code in this patch because it wasn't being used anywhere else.